<a href="https://colab.research.google.com/github/Arigalan/Practica_Merge/blob/main/Practica_merge.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
orders = pd.read_csv("olist_orders_dataset.csv")
customers = pd.read_csv("olist_customers_dataset.csv")
items = pd.read_csv("olist_order_items_dataset.csv")
products = pd.read_csv("olist_products_dataset.csv")

Parte 1 — Exploración de DataFrames

1. Mostrar dimensiones

Usa:

customers.shape

orders.shape

items.shape

products.shape

Para analizar las dimensiones de cada DataFrame.

2. Mostrar columnas

Usa:

orders.columns

customers.columns

items.columns

products.columns

Para identificar las columnas de cadaDataFrame.

In [ ]:
customers.shape

(11547, 5)

In [ ]:
orders.shape

(5904, 8)

In [ ]:
items.shape

(91624, 7)

In [ ]:
products.shape

(29048, 9)

In [ ]:
orders.columns

Index(['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date'],
      dtype='object')

In [ ]:
customers.columns

Index(['customer_id', 'customer_unique_id', 'customer_zip_code_prefix',
       'customer_city', 'customer_state'],
      dtype='object')

In [ ]:
items.columns

Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value'],
      dtype='object')

In [ ]:
products.columns

Index(['product_id', 'product_category_name', 'product_name_lenght',
       'product_description_lenght', 'product_photos_qty', 'product_weight_g',
       'product_length_cm', 'product_height_cm', 'product_width_cm'],
      dtype='object')

Parte 2 — Merge

3. Crear orders_customers

Realiza un merge entre:

orders

customers

Usando la columna:

customer_id

Guarda el resultado en:

orders_customers

In [ ]:
orders_customers = pd.merge(
    orders,
    customers,
    on = "customer_id",
    how = "inner"
)
orders_customers

,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00,3a653a41f6f9fc3d2a113cf8398680e8,75265,vianopolis,GO
1,82566a660a982b15fb86e904c8d32918,d3e3b74c766bc6214e0c830b17ee2341,delivered,2018-06-07 10:06:19,2018-06-09 03:13:12,2018-06-11 13:29:00,2018-06-19 12:05:52,2018-07-18 00:00:00,e97109680b052ee858d93a539597bba7,35400,ouro preto,MG
2,dcb36b511fcac050b97cd5c05de84dc3,3b6828a50ffe546942b7a473d70ac0fc,delivered,2018-06-07 19:03:12,2018-06-12 23:31:02,2018-06-11 14:54:00,2018-06-21 15:34:32,2018-07-04 00:00:00,ccafc1c3f270410521c3c6f3b249870f,74820,goiania,GO
3,116f0b09343b49556bbad5f35bee0cdf,3187789bec990987628d7a9beb4dd6ac,delivered,2017-12-26 23:41:31,2017-12-26 23:50:22,2017-12-28 18:33:05,2018-01-08 22:36:36,2018-01-29 00:00:00,6087cfc70fd833cf2db637a5e6e9d76b,88780,imbituba,SC
4,83018ec114eee8641c97e08f7b4e926f,7f8c8b9c2ae27bf3300f670c3d478be8,delivered,2017-10-26 15:54:26,2017-10-26 16:08:14,2017-10-26 21:46:53,2017-11-08 22:22:00,2017-11-23 00:00:00,634f09f6075fe9032e6c19609ffe995a,44024,feira de santana,BA
...,...,...,...,...,...,...,...,...,...,...,...,...
702,6a1ac461a015123f2d4daef591933747,b19151cf03d1ee88b38e392fac8acc62,delivered,2017-12-07 15:57:15,2017-12-08 02:53:39,2017-12-18 13:06:33,2017-12-28 19:52:18,2018-01-12 00:00:00,9041aaf6094cf78b32f85afbdcdc9721,78590,paranaita,MT
703,3c7bbd575f4e18f9fb416be0cb30840b,492465ef3d63cce9b19ee76f52ac8037,delivered,2018-03-23 22:20:08,2018-03-23 22:30:31,2018-03-27 00:06:38,2018-03-29 20:03:15,2018-04-11 00:00:00,c011547921af20560248edbfa600c4f4,6763,taboao da serra,SP
704,217f784e30b599df959cb81522d68f73,61ca21aafb7110bf6cd2078f01d7a6e5,delivered,2017-12-18 21:38:35,2017-12-22 12:11:48,2017-12-26 16:54:55,2018-01-03 18:29:04,2018-01-12 00:00:00,465213418e5f34776c10da6c25583be5,70294,brasilia,DF
705,56c68bdc0271f13c3037a5bb26d520ad,28b19e4aa3b8984663064c407ca727ff,delivered,2017-06-22 22:44:21,2017-06-22 22:55:12,2017-06-26 08:49:32,2017-07-02 10:07:46,2017-07-12 00:00:00,e6615a6b892c0ae58f8d1e1bd8f1c609,2402,sao paulo,SP


4. Analizar columnas

Muestra las columnas de:

orders_customers

y analiza:

Qué columnas pertenecían originalmente a orders

Qué columnas pertenecían originalmente a customers

Qué columnas fueron agregadas después del merge

In [ ]:
orders_customers.columns

Index(['order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date',
       'customer_unique_id', 'customer_zip_code_prefix', 'customer_city',
       'customer_state'],
      dtype='object')

¿Qué columnas pertenecían originalmente a orders?


R:Las columnas que pertenecian orignialmente a orders eran:

'order_id', 'customer_id', 'order_status', 'order_purchase_timestamp',
       'order_approved_at', 'order_delivered_carrier_date',
       'order_delivered_customer_date', 'order_estimated_delivery_date'

¿Qué columnas pertenecían originalmente a customers?

R: Las columnas que pertenecian originalmente a customers eran:

'customer_id', 'customer_unique_id', 'customer_zip_code_prefix',
       'customer_city', 'customer_state'

¿Qué columnas fueron agregadas después del merge?

R: Se unieron las columnas de orders y de customers

5. Crear items_products

Realiza un merge entre:

items

products

usando la columna:

product_id

Guarda el resultado en:

items_products

In [ ]:
items_products = pd.merge(
    items,
    products,
    on = "product_id",
    how = "inner"
)
items_products

,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.90,13.29,cool_stuff,58.0,598.0,4.0,650.0,28.0,9.0,14.0
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,2017-05-03 11:05:13,239.90,19.93,pet_shop,56.0,239.0,2.0,30000.0,50.0,30.0,40.0
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,2018-01-18 14:48:30,199.00,17.87,moveis_decoracao,59.0,695.0,2.0,3050.0,33.0,13.0,33.0
3,00024acbcdf0a6daa1e931b038114c75,1,7634da152a4610f1595efa32f14722fc,9d7a1d34a5052409006425275ba1c2b4,2018-08-15 10:10:18,12.99,12.79,perfumaria,42.0,480.0,1.0,200.0,16.0,10.0,15.0
4,00042b26cf59d7ce69dfabb4e55b4fd9,1,ac6c3623068f30de03045865e4e10089,df560393f3a51e74553ab94004ba5c87,2017-02-13 13:57:51,199.90,18.14,ferramentas_jardim,59.0,409.0,1.0,3750.0,35.0,40.0,30.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
80009,cfe6280cb115ade27e379ed940db3f1f,1,3a4db42aea95097b7a616b36f21d143f,bfd27a966d91cfaafdb25d076585f0da,2018-01-11 20:30:32,210.00,17.23,papelaria,56.0,225.0,1.0,1000.0,33.0,32.0,38.0
80010,cfe75b350e0f6cd144578e60bd77da82,1,5d66715cc928aadd0074f61332698593,128639473a139ac0f3e5f5ade55873a5,2018-04-19 16:30:37,19.90,18.23,eletronicos,52.0,390.0,1.0,150.0,20.0,13.0,18.0
80011,cfe7b4c34db01ec6e038ce0d1f4e75c9,1,ade881d3a9e73b511c630460c74dd92f,85d9eb9ddc5d00ca9336a2219c97bb13,2018-03-16 10:55:37,27.90,16.79,informatica_acessorios,31.0,1078.0,2.0,200.0,39.0,2.0,30.0
80012,cfe7e131a001d6fa542824cdc23399fe,1,72c0ede833dac61a24ac9a7c394066a5,3d871de0142ce09b7081e2b9d1733cb1,2018-04-26 09:31:04,28.50,19.32,papelaria,53.0,215.0,2.0,1250.0,38.0,30.0,28.0


In [ ]:
items_products.columns

Index(['order_id', 'order_item_id', 'product_id', 'seller_id',
       'shipping_limit_date', 'price', 'freight_value',
       'product_category_name', 'product_name_lenght',
       'product_description_lenght', 'product_photos_qty', 'product_weight_g',
       'product_length_cm', 'product_height_cm', 'product_width_cm'],
      dtype='object')

Preguntas:

1. ¿Qué utilidad tuvo unir orders con customers?

R: Porque ahora, con el join de las columnas de orders y de customers, no solamente podemos obtener los detalles de las órdenes, sino que también los datos del cliente, lo cual noas podría favorecer después para hacer un análisis estadístico.


2. ¿Qué información nueva se obtuvo después de crear orders_customers?

R: Se agregaron los datos que tenían en común orders y customers y se agregaron los datos del cliente.

3. ¿Qué utilidad tuvo unir items con products?

R: Que ahora con éste join podemos relacionar los productos que se venden con las características y especificaciones de los mismos.


4. ¿Qué diferencias observas entre los DataFrames originales y los DataFrames resultantes?

R: Puedo observar algunas diferencias, por ejemplo, en los dataframes originales, el total de columnas de orders era de 11547 y la de customers 5904, sin embargo, las columnas del orders_customers son de 707. Pasa los mismo con el join de items_products. Además, la información está más completa, debido a que se agrega información que antes cada DataFrame por separado no tenía.

5. ¿Qué columna funcionó como relación entre orders y customers?

R: La columna que funcionó como primary-key, como se ve en sql, o la que relaciona las dos tablas, fue "customer_id"


6. ¿Qué columna funcionó como relación entre items y products?

R: Fue la de product_id


7. ¿Para qué podría servir el DataFrame orders_customers en un análisis real?

R: Como comenté anteriormente, este DataFrame puede servir para hacer un análisis estadístico de quizás por que un cliente tiene tan pocas ordenes y como podriamos maximizar la venta de un cierto grupo de clientes viendo sus datos de compras y de cliente como tal.


8. ¿Para qué podría servir el DataFrame items_products en un análisis real?

R: Puede servir quizá para hacer un análisis de si el precio del producto está bien en relación a sus características.